lib/util/Components.js

```diff
@@ -486,6 +492,23 @@ function componentRule(rule, context) {
       return null;
     },
 
+    /**
+     * Get a parent scope created by a FunctionExpression or FunctionDeclaration
+     * @param {Scope} scope The child scope
+     * @returns {Scope} A parent function scope
+     */
+    getParentFunctionScope(scope) {
+      scope = scope.upper;
+      while (scope) {
+        const type = scope.block.type;
+        if (type === 'FunctionExpression' || type === 'FunctionExpression') {
```

Comment:
Is this meant to be checking the same thing twice?

Reply:
good catch, it should be `FunctionExpression` and `FunctionDeclaration`